### DISCUSSION TITLE
CodeBlock render code without theme

### DISCUSSION THREAD

**Discussion by Ali-Hussein-dev** [fuma-nama/fumadocs#867] [Q&A] [answered]
Need to pass dynamic content to `CodeBlock` but the code snippet lose the theme, 
```tsx
import { getCodeBlock } from '@/lib/code-block';
import { Pre, CodeBlock } from 'fumadocs-ui/components/codeblock';

export async function CodeSnippet() {
  const code = await getCodeBlock('/src/components/footer.tsx');
  return (
    <CodeBlock>
      <Pre>{code.body}</Pre>
    </CodeBlock>
  );
}
````

```mdx
// mdx file
import {CodeSnipeet} from "...."
<CodeSnippet />
```

The code snippet looks like this and I haven't touched MDX component.
<img width="935" alt="image" src="https://github.com/user-attachments/assets/0e25b3d0-e4bc-403c-836d-f9ea59811d4a">

**Comment by fuma-nama** [comment, marked as answer]
Hmm guess you misunderstood what the component does. It styles the output of Rehype Code (our tiny wrapper over Shiki), and you are supposed to use it with Markdown/MDX.

Your usage requires syntax highlighting with Shiki and render the Hast tree with code block component.

I would suggest you to take a look at our source code of landing page, or see Shiki's Next.js integration guide (as I wrote it, I can confirm it's correct)

**Reply by Ali-Hussein-dev** [reply]
Can you share the link to Shiki's Next.js integration guide?

**Reply by fuma-nama** [reply]
See their docs ;)
